server/src/app-dto/enter-results.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  ArrayMaxSize,
+  ArrayMinSize,
+  IsInt,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+  Min,
+  Validate,
+  ValidateNested,
+} from 'class-validator';
+import { AttemptDto, HasNonDnsResult } from '../modules/results/dto/create-result.dto';
+import { Type } from 'class-transformer';
+import { IAttempt } from '~~/client/shared_helpers/types';
+
+export class EnterResultsDto {
+  @IsString()
+  @IsNotEmpty()
+  competitionWcaId: string;
+
+  @IsString()
+  @IsNotEmpty()
+  eventId: string;
+
+  @IsNumber()
+  roundNumber: number;
+
+  @ValidateNested({ each: true })
+  results: ExternalResultDto[];
```

Comment:
Did you test that this works and does the same thing as when you use @Type(() => ExternalResultDto)?

Reply:
I've added this and tested some scenarios, but I didn't see any difference. In which cases does using `@Type(() => ExternalResultDto)` make a difference compared to not using it?

Reply:
Oh, if it doesn't make a difference, you can change it back. I never thought to use the DTO as the type directly, I just used an interface and then specified the DTO using @type. I might end up changing it to your syntax everywhere else actually.